include/deal.II/lac/generic_linear_algebra.h

```diff
@@ -149,74 +516,206 @@ DEAL_II_NAMESPACE_OPEN
+
+      /**
+       * Typedef for the direct linear solver.
+       */
+      typedef TrilinosWrappers::SolverDirect SolverDirect;
+
+      /**
+       * Typedef for the GMRES linear solver.
+       */
+      typedef TrilinosWrappers::SolverGMRES SolverGMRES;
+
+      /**
+       * Typedef for the TFQMR linear solver.
+       */
+      typedef TrilinosWrappers::SolverTFQMR SolverTFQMR;
+    }
+
+    namespace CommonPreconditioners
+    {
+      /**
+       * Typedef for the base class for preconditioners.
+       *
+       * This typedef is useful when wanting to create a generic solver interface
+       * wherein one can switch between preconditioners. For example:
+       * @code
+       * std_cxx11::unique_ptr<LA::PreconditionBase> preconditioner;
+       * if (parameters.preconditioner_type == "Jacobi")
+       * {
+       *   LA::MPI::PreconditionJacobi* ptr_prec
+       *     = new LA::MPI::PreconditionJacobi ();
+       *
+       *   LA::MPI::PreconditionJacobi::AdditionalData
+       *     additional_data (parameters.preconditioner_relaxation);
+       *
+       *   ptr_prec->initialize(tangent_matrix.block(u_block,u_block),
+       *                        additional_data);
+       *   preconditioner.reset(ptr_prec);
+       * }
+       * @endcode
+       */
+      typedef TrilinosWrappers::PreconditionBase PreconditionBase;
+
+      /**
+       * Typedef for the AMG preconditioner type.
+       */
+      typedef TrilinosWrappers::PreconditionAMG PreconditionAMG;
+
+      /**
+       * Typedef for the Incomplete Cholesky preconditioner.
+       */
+      typedef TrilinosWrappers::PreconditionIC PreconditionIC;
+
+      /**
+       * Typedef for the identity preconditioner.
+       */
+      typedef TrilinosWrappers::PreconditionIdentity PreconditionIdentity;
+
+      /**
+       * Typedef for the Incomplete LU decomposition preconditioner.
+       */
+      typedef TrilinosWrappers::PreconditionILU PreconditionILU;
+
+      /**
+       * Typedef for the Incomplete Jacobi decomposition preconditioner.
+       */
+      typedef TrilinosWrappers::PreconditionJacobi PreconditionJacobi;
+
+      /**
+       * Typedef for the SOR preconditioner
+       */
+      typedef TrilinosWrappers::PreconditionSOR PreconditionSOR;
+
+      /**
+       * Typedef for the SSOR preconditioner
+       */
+      typedef TrilinosWrappers::PreconditionSSOR PreconditionSSOR;
+    }
+  } // LinearAlgebra::Trilinos::internal
+
+  /**
+   * Typedef for the vector type used.
+   */
+  typedef dealii::TrilinosWrappers::Vector Vector;
+
+  /**
+   * Typedef for the type used to describe vectors that consist of multiple
+   * blocks.
+   */
+  typedef dealii::TrilinosWrappers::BlockVector BlockVector;
 
-  typedef TrilinosWrappers::SolverCG SolverCG;
-  typedef TrilinosWrappers::SolverGMRES SolverGMRES;
+  // Incorporate common elements into namespace
+  using namespace LinearAlgebraTrilinos::internal::CommonLinearAlgebra;
```

Comment:
I'm not sure you can. The contents of that namespace continue to live there, they just have been made visible in the current namespace. 

If you know the contents of the imported namespace, you can of course bring every member of that namespace to the current one via a typedef, and that should be visible in doxygen.